File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/VehicleComponent.java
package com.example.tutorial4_1dependentcomponent.di;

import com.example.tutorial4_1dependentcomponent.MainActivity;
import com.example.tutorial4_1dependentcomponent.model.Vehicle;

import javax.inject.Singleton;

import dagger.Component;
@Component(dependencies = {MotorComponent.class}, modules = {VehicleModule.class})
public interface VehicleComponent {
    void inject(MainActivity mainActivity);
}

File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/VehicleComponent.kt
package com.example.tutorial4_1dependentcomponent.di

import com.example.tutorial4_1dependentcomponent.MainActivity
import dagger.Component

@Component(dependencies = [MotorComponent::class], modules = [VehicleModule::class])
interface VehicleComponent {

    fun inject(mainActivity: MainActivity)

}